File: all-submissions/0362-design-hit-counter/solution.java
// class HitCounter {
//     private Queue<Integer> hits;

//     public HitCounter() {
//         this.hits = new LinkedList<Integer>();
//     }
    
//     public void hit(int timestamp) {
//         this.hits.add(timestamp);
        
//     }
    
//     public int getHits(int timestamp) {
//         while(!this.hits.isEmpty()){
//             int diff = timestamp - this.hits.peek();
//             if (diff >= 300) this.hits.remove();
//             else break;
//         }
//         return this.hits.size();
        
//     }
// }

/**
 * Your HitCounter object will be instantiated and called as such:
 * HitCounter obj = new HitCounter();
 * obj.hit(timestamp);
 * int param_2 = obj.getHits(timestamp);
 */



 class HitCounter {
     Queue<Integer> hitQueue;

     HitCounter() {
         hitQueue = new LinkedList<>();
     }

     public void hit(int timestamp) {
         hitQueue.add(timestamp);
     }

     public int getHits(int timestamp) {
         //remove all queue elements more than 300s old 
        

        while(!hitQueue.isEmpty()) {
            int firstElement = hitQueue.peek();
            if ((timestamp - firstElement) >= 300 ) {
                hitQueue.remove();
            } else {
                break;
            }
        }

         return hitQueue.size();
     }
 }

File: all-submissions/0362-design-hit-counter/solution.py
class HitCounter:

    def __init__(self):
        self.hit_counter = defaultdict(int)

    def hit(self, timestamp: int) -> None:
        self.hit_counter[timestamp] +=1

    def getHits(self, timestamp: int) -> int:
        totalCount = 0
        diff = timestamp-300 if timestamp >= 300 else 0
        for t in self.hit_counter.keys():
            if diff < t <= timestamp:
                totalCount += self.hit_counter[t]
        return totalCount
            


# Your HitCounter object will be instantiated and called as such:
# obj = HitCounter()
# obj.hit(timestamp)
# param_2 = obj.getHits(timestamp)
